driver-sync/src/test/functional/com/mongodb/client/ClientSideEncryptionProseTestSpecification.groovy

```diff
@@ -0,0 +1,480 @@
+
+        then:
+        decryptedValue == value
+    }
+
+    /*
+Test explicit encrypt of invalid values.
+
+- Create a `KeyVault` with either a "local" or "aws" KMS provider
+- Use `KeyVault.encrypt` to attempt to encrypt each BSON type with deterministic encryption.
+
+  - Expect `document` and `array` to fail. An exception MUST be thrown.
+  - Expect a BSON binary subtype 6 to fail. An exception MUST be thrown.
+  - Expect all other values to succeed.
+
+- Use `KeyVault.encrypt` to attempt to encrypt a document using randomized encryption.
+
+  - Expect a BSON binary subtype 6 to fail. An exception MUST be thrown.
+  - Expect all other values to succeed.
+
+ */
+
+    def 'should encrypt valid values'() {
+        given:
+        dataKeyCollection.insertOne(localDataKeyDocument)
+        def keyVault = KeyVaults.create(KeyVaultEncryptionSettings.builder()
+                .keyVaultMongoClientSettings(getMongoClientSettings())
+                .keyVaultNamespace(keyVaultNamespace.fullName)
+                .kmsProviders(localProviderProperties)
+                .build())
+        def options = new EncryptOptions(algorithm).keyId(localDataKeyDocument.getBinary('_id'))
+
+        when:
+        keyVault.encrypt(value, options)
+
+        then:
+        noExceptionThrown()
+
+        where:
+        [value, algorithm] << [
+                [new BsonInt32(1),
+                 new BsonInt64(1L),
+                 new BsonTimestamp(42),
+                 new BsonObjectId(new ObjectId()),
+                 new BsonBinary(new byte[4]),
+                 new BsonString('42'),
+                 new BsonSymbol('42')],
+                ['AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic',
+                 'AEAD_AES_256_CBC_HMAC_SHA_512-Random']
+        ].combinations()
+    }
+
+    def 'should fail to encrypt invalid values with deterministic encryption'() {
+        given:
+        dataKeyCollection.insertOne(localDataKeyDocument)
+        def keyVault = KeyVaults.create(KeyVaultEncryptionSettings.builder()
+                .keyVaultMongoClientSettings(getMongoClientSettings())
+                .keyVaultNamespace(keyVaultNamespace.fullName)
+                .kmsProviders(localProviderProperties)
+                .build())
+        def options = new EncryptOptions('AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic')
+                .keyId(localDataKeyDocument.getBinary('_id'))
+
+        when:
+        keyVault.encrypt(BsonNull.VALUE, options)
+
+        then:
+        thrown(MongoCryptException)
+
+        when:
+        keyVault.encrypt(BsonBoolean.TRUE, options)
+
+        then:
+        thrown(MongoCryptException)
+
+        when:
+        keyVault.encrypt(new BsonDouble(1.0), options)
+
+        then:
+        thrown(MongoCryptException)
+
+        when:
+        keyVault.encrypt(new BsonDecimal128(Decimal128.parse('1.0')), options)
+
+        then:
+        thrown(MongoCryptException)
+
+        // TODO: enable this
```

Comment:
There's a libmongocrypt bug that is making this fail.  Once the fix is in master this can be enabled